packages/next/src/server/dev/next-dev-server.ts

```diff
@@ -555,7 +555,7 @@ export default class DevServer extends Server {
               continue
             }
             // Ignore files/directories starting with `_` in the app directory
-            if (normalizePathSep(fileName).includes('/_')) {
+            if (normalizePathSep(pageName).includes('/_')) {
```

Comment:
We use the `pageName` instead of the `fileName` because the `fileName` includes path parts outside the Next.js directory.